content/blog/federating-spire-on-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,447 @@
+
+## 2.3 Patch Istio Ingress Gateway
+
+### 2﻿.3.1 Apply SPIFFE ID
+
+First, you must get and apply one of SPIRE controller manager’s [CRD (Custom Resource Definition)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) ClusterSPIFFEID. The CRD - ClusterSPIFFEID is a cluster-wide resource used to register workloads with SPIRE. The ClusterSPIFFEID can target all workloads in the cluster or can be optionally scoped to specific pods or namespaces via label selectors.  
+
+Create a ClusterSPIFFEID CRD to generate registration entries in SPIRE server for all workloads labeled ***spiffe.io/spire-managed-identity: true.***
+
+Get the ClusterSPIFFEID used for this demo using this [link](https://github.com/cxteamtrials/caas-trials-content/blob/main/services/spire/clusterspiffeid-example.yaml) and  apply it to both clusters. 
+
+```shellsession
+kubectl apply -f services/spire/clusterspiffeid-example.yaml 
+```
+
+### 2.3﻿.2 Patch Ingress Gateway
+
+Now, simply patch the ingress-gateway with ***spiffe.io/spire managed-identity: true label.***
+
+This patch will register your ingress-gateway pod into the server.
+
+```shellsession
+kubectl patch deployment istio-ingressgateway -n istio-system -p '{"spec":{"template":{"metadata":{"labels":{"spiffe.io/spire-managed-identity": "true"}}}}}' 
+```
+
+### 2﻿.3.3 Check your work
+
+After patching, confirm that your ingress-gateway pod, istiod and all their containers work.
+
+```shellsession
+Cluster1:~ # kubectl get po -n istio-system 
+NAME                                   READY   STATUS    RESTARTS   AGE
+istio-ingressgateway-5d77cdd9d-gh9w4   1/1     Running   0          37d
+istiod-d5bc8669c-4bdvh                 1/1     Running   0          37d
+```
+
+```shellsession
+Cluster2:~ #  kubectl get po -n istio-system 
+NAME                                   READY   STATUS    RESTARTS   AGE
+istio-ingressgateway-64bd5ccbbb-kqs2h  1/1     Running   0          37d
+istiod-d5bc8669c-thbpj                 1/1     Running   0          37d
+```
+
+# Step 3. Federating SPIRE
+
+## 3﻿.1 Expose SPIRE server bundle endpoint
+
+Assign an external IP to your spire-server-bundle-endpoint service on each cluster.  
+
+SPIFFE (*Secure Production Identity Framework For Everyone*) is a specification for implementing identity for workloads, and SPIRE is the code that implements this specification in practice. A SPIFFE bundle is a resource that contains the public key material needed to authenticate credentials from a particular trust domain. A SPIFFE bundle endpoint is a resource (represented by a URL) that serves a copy of a SPIFFE bundle for a trust domain. SPIFFE control planes may both expose and consume these endpoints to transfer bundles between themselves, thereby achieving federation. The SPIRE server is used to host the “spire-server-bundle-endpoint” service that serves the SPIFFE bundle to an external Spire agent of a different trust domain.  
```

Comment:
Last sentence:
...to an external SPIRE agent of a different trust domain.

For consistency, I suggest spelling SPIRE agent with "SPIRE" in capital letters.